greedy_approach/greedy_coin_change.c

```diff
@@ -0,0 +1,172 @@
+/**
+ * @file
+ * @author [serturx](https://github.com/serturx)
+ * @brief This is a C implementation of the [greedy coin change
+ * algorithm](https://en.wikipedia.org/wiki/Change-making_problem)
+ * @details This algorithm computes the needed coins from a given set of coins
+ * that make up a given amount. It works greedy, so it chooses the biggest
+ * possible coin that fits in the remaining amount until the remaining sum is
+ * equal to 0 or no coin fits in the remaining amount.
+ */
+
+#include <stdlib.h>  /// for qsort
+#include <stdbool.h> /// for bool type
+#include <string.h>  /// for memcmp
+#include <assert.h>  /// for assert
+#include <stdio.h>   /// for IO operations
+
+/**
+ * @brief A struct containing the result of the coin change algorithm.
+ * It consists of the length of the result and the result itself (as an integer array)
+ * 
+ */
+typedef struct {
+    size_t result_length;
+    int *result;
```

Comment:
Please add a one-line description of what the variable is for (**see the example below**).

```c
int64_t data = 0;       ///< The value/key of the node.
```